libs/crypto/keys_manager_impl.cpp

```diff
@@ -69,39 +47,25 @@ namespace iroha {
   /**
    * Function for XOR decryption
    */
-  static constexpr auto decrypt = encrypt<std::string>;
-
-  /**
-   * Return a function which will try to deserialize and then decrypt private
-   * key via XORing with pass phrase
-   * @param s is an encrypted data from file
-   * @param pass_phrase for decryption
-   * @return function that will set keypair::privkey on successful
-   *         deserialization and decryption
-   */
-  static auto deserializedEncrypted(const std::string &s,
-                                    const std::string &pass_phrase) {
-    constexpr auto size = privkey_t::size();
-
-    return [=](auto keypair) mutable {
-      return hexstringToBytestring(s) |
-          [&](auto binstr) {
-            return boost::make_optional(decrypt(binstr, pass_phrase));
-          }
-      | stringToBlob<size> | assignObjectField(keypair, &keypair_t::privkey);
-    };
-  }
+  static constexpr auto decrypt = encrypt<shared_model::crypto::Blob::Bytes>;
 
   KeysManagerImpl::KeysManagerImpl(const std::string &account_name)
       : account_name_(std::move(account_name)),
         log_(logger::log("KeysManagerImpl")) {}
 
-  bool KeysManagerImpl::validate(const keypair_t &keypair) const {
-    std::string test = "12345";
-    auto sig =
-        sign(sha3_256(test).to_string(), keypair.pubkey, keypair.privkey);
-    if (not verify(sha3_256(test).to_string(), keypair.pubkey, sig)) {
-      log_->error("key validation failed");
+  bool KeysManagerImpl::validate(
+      const shared_model::crypto::Keypair &keypair) const {
+    try {
+      auto test = shared_model::crypto::Blob(sha3_256("12345").to_string());
+      auto sig =
+          shared_model::crypto::DefaultCryptoAlgorithmType::sign(test, keypair);
+      if (not shared_model::crypto::DefaultCryptoAlgorithmType::verify(
+              sig, test, keypair.publicKey())) {
+        log_->error("key validation failed");
+        return false;
+      }
+    } catch (const BadFormatException &exception) {
```

Comment:
Where is the exception may be throwed?

Reply:
Here https://github.com/hyperledger/iroha/blob/9e0e19035308c6ebaf706f709c5b7b3ac46e708b/libs/common/types.hpp#L110
from https://github.com/hyperledger/iroha/blob/9e0e19035308c6ebaf706f709c5b7b3ac46e708b/shared_model/cryptography/ed25519_sha3_impl/signer.cpp#L29

Reply:
Nice catch! Nevertheless it seems redundant after your fixes

Reply:
Yes, but i don't want exception to be uncaught.